addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -650,7 +651,22 @@ var SnippetEditor = Widget.extend({
         // because some options can be located in the overlay.
         const $visibleOptions = this.$optionsSection.find('we-top-button-group, we-customizeblock-option')
                 .children(':not(.d-none)');
-        this.$optionsSection.toggleClass('d-none', !$visibleOptions.length);
+        // Some options (e.g., text highlights / animations) may have a special
+        // way to be displayed in the editor: We add the options in the toolbar
+        // `onFocus()` and set them back `onBlur()`. Which means that the
+        // options section will be empty and should be hidden, while editor's
+        // visible options should be displayed in the toolbar DOM. We need to
+        // take this scenario into consideration too.
+        const optionsSectionEmpty = !this.$optionsSection[0].querySelector(":scope > we-customizeblock-option");
+        const optionsSectionVisible = $visibleOptions.length && !optionsSectionEmpty;
+        // At this level, we can hide the options section.
+        this.$optionsSection.toggleClass("d-none", !optionsSectionVisible);
+        // Even with a hidden options section, the editor is still considered
+        // visible" if it has visible toolbar options.
+        const visibleToolbarOptions = Object.values(this.styles).some(
+            (option) => option.el.closest(".oe-toolbar") && !option.el.classList.contains("d-none")
+        );
```

Comment:
I still don't understand that part: at the top of this `updateOptionsUIVisibility`, we call updateUIVisibility for each option -> cannot this also resolve to true or false depending on if the option is now visible/hidden? That way would it still be needed to check .oe-toolbar and others by hand?

Reply:
Yes, I think this is possible. I added an implementation in a separate commit here since I'm not sure 100% if the `displayOverlayOptions` check is robust enough to indicate that an option is in the overlay to exclude it... If it seems OK for you, just let me know and I will squash it :+1: